### ISSUE TITLE
Comments: trouble saving comments while on bad network connections

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/WordPress-iOS#11307]
Steps to reproduce:

<ol>
<li>Go to Notifications and tap on any comment.</li>
<li>Throttle the network connection.*</li>
<li>Write a reply and tap "REPLY".</li>
<li>See a notice that says "There has been an unexpected error while sending your reply."</li>
<li>Tap "Try Again".</li>
<li>Return to a good network connection.</li>
<li>Check to see whether the comment posted.</li>
</ol>

Result: comment fails to post silently and is lost.

<img src="https://user-images.githubusercontent.com/1119271/54724980-f60ce900-4b32-11e9-9606-575b9d8b2af3.PNG" width="270" alt="IMG_2163"> <img src="https://user-images.githubusercontent.com/1119271/54724998-07ee8c00-4b33-11e9-827c-00b94e968d39.PNG" width="270" alt="IMG_2164">

<em>Tested with WP Internal 12.0.0.20190314 on iPhone 6S iOS 12.1.4 while replying to a comment on <code>madefortesting190318.wordpress.com</code> which is a WordPress.com Free site.</em>

<sub>* I throttled my network connection by going into a back room in my place where the wifi doesn't reach and is terrible. I don't have phone service connected to my iPhone so I couldn't do a wifi-to-cellular-network switch as part of the test but that may be worth trying if you have difficulty replicating the bug.</sub>

h/t @mattmiklic for the original report at p4a5px-2mi-p2.

<blockquote>Got a weird bug: tried to leave a comment on bad wifi, it just silently wasn’t working. Turned off WiFi to switch to LTE<br>
Got a retry screen I missed<br>
And this<br>
<img width="138" alt="screen-shot-2019-03-20-at-10 40 05-am" src="https://user-images.githubusercontent.com/1119271/54724955-dd043800-4b32-11e9-806a-6e12ce3336fe.png"><br>
The comment appears to have been totally lost.</blockquote>

**Comment by designsimply** [comment]
I was re-checking this and I found that the comment in question actually did go through eventually at https://madefortesting190318.wordpress.com/2019/03/20/yoga-for-your-back/comment-page-1/#comment-2 ! 😱

However, the original report still stands and needs to be checked:

<blockquote>Got a weird bug: tried to leave a comment on bad wifi, it just silently wasn’t working. Turned off WiFi to switch to LTE<br>
Got a retry screen I missed<br>
And this<br>
<img width="138" alt="screen-shot-2019-03-20-at-10 40 05-am" src="https://user-images.githubusercontent.com/1119271/54724955-dd043800-4b32-11e9-806a-6e12ce3336fe.png"><br>
The comment appears to have been totally lost.</blockquote>

**Comment by yaelirub** [comment]
Been trying to reproduce this issue all day and succeeded twice and only in a reply to a reply on a comment.
![IMG_3586](https://user-images.githubusercontent.com/1335657/54966929-88a6f100-4f54-11e9-94a2-2f1143405106.PNG)

Working with @designsimply to try and reproduce more consistently.

**Comment by designsimply** [comment]
I tried the same steps again, over a longer time period, trying to replicate losing a comment when posting via Notifications from a bad network connection and here's what I found:

I logged in as `designsimply` and using WPiOS 11.9.1 on iPhone 6S iOS 12.1.4, I left two comments last night as replies. Then I left the phone on and in an area with bad wifi and left the phone on until the morning. 

For the [first comment](https://mobilerequests.wordpress.com/2019/03/21/unable-to-upload-images-in-post/#comment-9579), I never saw the "Try Again" prompt and the comment has not appeared so far:

<img src="https://user-images.githubusercontent.com/1119271/55025394-79f92200-4fc6-11e9-817a-c4dcf2cd1a3a.PNG" width="270" alt="IMG_2179">

For the [second comment](https://apptestingo2.wordpress.com/2019/03/11/to-test-wordpress-android-12-0/#comment-2632), I did see the "Try Again" prompt and tapped it and the comment published at that time with a timestamp on Mar 26 even though I created and sent the comment on Mar 25.

<img src="https://user-images.githubusercontent.com/1119271/55025650-fbe94b00-4fc6-11e9-91c4-d7c7b9597bab.PNG" width="270" alt="IMG_2181">  <img src="https://user-images.githubusercontent.com/1119271/55025720-1e7b6400-4fc7-11e9-895c-8754f50599ee.PNG" width="270" alt="IMG_2184">

@yaelirub perhaps commenting multiple times in a row from a truly bad connection will result in the first comments from the session failing while the latest ones go through?

**Comment by yaelirub** [comment]
This is a difficult one to reproduce consistently but I think I got it (asking @designsimply to confirm).
1. Turn off cellular service 
2. Turn NLC (Network Link Conditioner) on 100% loss
3. Reply to a reply on a comment
4. turn off wifi completely.
5. You will see the “There was a problem” alert.
6. Choose cancel or try again for that matter.
7. Turn wifi on (NLC off)
8. The comment is lost. (edited) 
*Sometimes the NLC stops working and so you have to turn it off, turn wifi off and on and turn the NLC on again.

I agree with @designsimply saying that the fix for this issue feels like part of a project work. 
But I have a couple of questions, @elibud :
1. The original report states that the "There has be an unexpected error..." alert was missed. If the user hits cancel on this alert do we expect the comment to be deleted? Does cancel mean canceling the comment (as oppose to "try again")?
2. Why are we even showing this alert? Can we remove it? imo it's not clear and the 2 options are confusing. We should "try again" for the user.

**Comment by elibud** [comment]
Yes, that alert is definitely confusing. If there is a common pattern we are using somewhere else let's update it to follow it. If not let's make sure we address issues like this one in the offline project.

**Comment by yaelirub** [comment]
https://github.com/wordpress-mobile/WordPress-iOS/pull/11817 address improving the messaging experience when replying/ commenting. This issue still remains and will be tackled in Offline support.
